### DISCUSSION TITLE
alloca - variable length stack allocation

### DISCUSSION THREAD

**Discussion by ntrel** [vlang/v#11737] [Ideas]
It would be nice to have something like C's alloca: https://man7.org/linux/man-pages/man3/alloca.3.html.

alloca is not portable, but it could be supported just for C compilers that have it. C99 Variable Length Arrays is a standard feature but I think it was made an optional feature in later standards.

MSVC has [malloca](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/malloca?view=msvc-160) which requires freeing with `freea`. This combination could also solve portability issues because it can fall back to heap allocation on systems without runtime stack allocation.

**Comment by danieldaeschle** [comment]
This is nice, why not use this instead of autofree? :D

**Comment by k1attila12** [comment]
because it is on stack ?

**Comment by ntrel** [comment]
Submitted too early, updated description.
